src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js

```diff
@@ -106,15 +117,27 @@ $(function () {
         success: function (response) {
           debugResponse.text(response);
           hljs.highlightBlock(debugResponse.get(0));
+          var uri = URI(window.location.href);
+          var fragment = uri.fragment(true);
+          fragment.setQuery('req', debugText.val());
+          window.location.href = uri.toString();
         },
         error: function (jqXHR, textStatus, errorThrown) {
           debugResponse.text(errorThrown);
         }
       });
       return false;
-    });
+    };
+    functionContainer.find('.debug-submit').on('click', submitDebugRequest);
 
     functionContainer.removeClass('hidden');
+
+    var fragment = URI(window.location.href).fragment(true);
+    var fragmentParams = fragment.query(true);
+    if (fragmentParams.req) {
+      debugText.val(fragmentParams.req);
+      submitDebugRequest();
```

Comment:
Yeah - I feel the link is to the actual response and submitting by default is good behavior. Imagine a QA report with a debug link inside it - you click it to see what the response was to debug in pretty much all cases.